src/overlays/actors/ovl_Obj_Purify/z_obj_purify.c

```diff
@@ -27,42 +43,252 @@ const ActorInit Obj_Purify_InitVars = {
+        { object_numa_obj_Matanimheader_012CF8, object_numa_obj_Matanimheader_0131A8 },
+        &object_numa_obj_Colheader_012818,
+        0,
+    },
+    {
+        0x19A,
+        1.0f,
+        { NULL, NULL },
+        { object_dekucity_obj_DL_0004E0, object_dekucity_obj_DL_001030 },
+        { object_dekucity_obj_Matanimheader_000E28, object_dekucity_obj_Matanimheader_001978 },
+        &object_dekucity_obj_Colheader_0002AC,
+        1,
+    },
+    {
+        0x19A,
+        1.0f,
+        { object_dekucity_obj_DL_001ED8, object_dekucity_obj_DL_003998 },
+        { object_dekucity_obj_DL_001D80, NULL },
+        { object_dekucity_obj_Matanimheader_003820, NULL },
+        NULL,
+        1,
+    },
+    {
+        0x164,
+        0.1f,
+        { NULL, NULL },
+        { object_numa_obj_DL_012A60, object_numa_obj_DL_012F10 },
+        { object_numa_obj_Matanimheader_012B58, object_numa_obj_Matanimheader_013008 },
+        &object_numa_obj_Colheader_012818,
+        0,
+    },
+    {
+        0x164,
+        0.1f,
+        { NULL, NULL },
+        { object_numa_obj_DL_012BF0, object_numa_obj_DL_0130A0 },
+        { object_numa_obj_Matanimheader_012CE8, object_numa_obj_Matanimheader_013198 },
+        &object_numa_obj_Colheader_012818,
+        0,
+    },
+};
+
+void ObjPurify_SetSysMatrix(f32 ypos) {
+    Matrix_GetCurrent()->yw = ypos;
+}
+
+s32 ObjPurify_IsPurified(ObjPurify* this) {
+    ObjPurifyInfo* info;
+    info = &ObjPurifyInfoList[this->dyna.actor.params & 0xF];
+    if (!info->isDekuCity) {
+        // woodfall temple wood flower unraveled
+        if ((gSaveContext.save.weekEventReg[12] & 1)) {
+            return true;
+        }
+    }
+    // woodfall temple purification cutscene watched
+    else if (gSaveContext.save.weekEventReg[20] & 2) {
+        return true;
+    }
+    return false;
+}
+
+void ObjPurify_Init(Actor* thisx, PlayState* play) {
+    ObjPurify* this = THIS;
+    s32 pad;
+    ObjPurifyInfo* info;
+    s32 sp20;
+
+    info = &ObjPurifyInfoList[this->dyna.actor.params & 0xF];
+    sp20 = (this->dyna.actor.params >> 0xC) & 1;
+    Actor_SetScale(&this->dyna.actor, info->scale);
+    if (sp20 == 1) {
+        DynaPolyActor_Init((DynaPolyActor*)this, 0);
+    }
+    this->objIndex = Object_GetIndex(&play->objectCtx, info->objectId);
+    if (this->objIndex < 0) {
+        Actor_MarkForDeath(&this->dyna.actor);
+    } else if (sp20 == 0) {
+        func_80A84EAC(this);
+    } else if (ObjPurify_IsPurified(this)) {
+        Actor_MarkForDeath(&this->dyna.actor);
+    } else {
+        func_80A84EAC(this);
+    }
+}
+
+void ObjPurify_Destroy(Actor* thisx, PlayState* play) {
+    ObjPurify* this = THIS;
+    if (((this->dyna.actor.params >> 0xC) & 1) == 1) {
+        DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+    }
+}
+
+void func_80A84EAC(ObjPurify* this) {
+    this->actionFunc = func_80A84EC0;
+}
+
+void func_80A84EC0(ObjPurify* this, PlayState* play) {
+    s16 pad;
```

Comment:
Assuming this pad is required, style is for it to be s32: you shouldn't need others.